master/internal/task/allocation.go

```diff
@@ -407,15 +434,23 @@ func (a *Allocation) ResourcesAllocated(ctx *actor.Context, msg sproto.Resources
 		a.idleTimeoutWatcher.PreStart(ctx)
 	}
 
-	for cID, r := range a.resources {
-		if err := r.Start(ctx, a.logCtx, spec, sproto.ResourcesRuntimeInfo{
-			Token:        token,
-			AgentRank:    a.resources[cID].Rank,
-			IsMultiAgent: len(a.resources) > 1,
-		}); err != nil {
-			return fmt.Errorf("starting resources (%v): %w", r, err)
+	if !a.req.Restore {
+		token, err := a.db.StartAllocationSession(a.model.AllocationID)
```

Comment:
it kinda makes sense to it here because it's not needed earlier. maybe the content of this entire `if` could be a `Allocation.startResources` method? not much of a difference though.